core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/SplitBalanceHelper.java

```diff
@@ -304,11 +304,13 @@ protected void zkCleanup() {
           List<String> trafTables = ZKUtil.listChildrenNoWatch(zkw, zSplitBalPathNoSlash);
           List<String> hbaseTables = ZKUtil.listChildrenNoWatch(zkw, SplitBalanceHelper.zkTable);
           
-          for(String tableName : trafTables) {
+          if(trafTables != null && hbaseTables != null) {

```

Comment:
trafTables should be subset of hbaseTables.  If it is just a plain hbase table, do you think it should be the responsibility of this class to cleanup the zk nodes. If so, we might need to re-factor this code. If not, this change should be fine though it is bit more defensive.